drift/lib/src/runtime/query_builder/expressions/case_when.dart

```diff
@@ -1,30 +1,56 @@
-import 'package:meta/meta.dart';
-
 import '../query_builder.dart';
 
+/// A `CASE WHEN` expression (without base expression) in sqlite.
+class CaseWhenExpression<R extends Object> extends _CaseWhenExpression<R> {
+  /// Creates a `CASE WHEN` expression from the independent components.
+  CaseWhenExpression({
+    required Map<Expression, Expression<R>> when,
+    Expression<R>? orElse,
+  }) : super._(null, when.entries.toList(), orElse);
+}
+
+/// A `CASE WHEN` expression (with base expression) in sqlite.
+class CaseWhenExpressionWithBase<T extends Object, R extends Object>
```

Comment:
If we have this as a separate class, I think it shouldn't be exposed - the `caseMatch` extension suffices.

Reply:
`@internal` should me enough 👍🏻